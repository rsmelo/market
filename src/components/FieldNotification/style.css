.fieldNotification {
  position: absolute;
  top: 100%;
  line-height: 120%;
  font-size: 0.8em;
  left: 0;
  padding: 0;
}

.error {
  color: var(--color-error);
}

.success {
  color: var(--color-success);
}

