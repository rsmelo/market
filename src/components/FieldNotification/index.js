import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import style from './style.css'

const FieldNotification = ({ children, type }) => (
  <span className={classNames(style.fieldNotification, style[type])}>
    {children}
  </span>
)

FieldNotification.defaultProps = {
  type: 'error',
}

FieldNotification.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
  type: PropTypes.oneOf(['success', 'error']),
}

export default FieldNotification
