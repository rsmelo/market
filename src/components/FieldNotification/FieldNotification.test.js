import React from 'react'
import renderer from 'react-test-renderer'

import FieldNotification from './index'

describe('FieldNotification', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<FieldNotification>Test</FieldNotification>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
