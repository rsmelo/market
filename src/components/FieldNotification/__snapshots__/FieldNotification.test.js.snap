// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldNotification should render correctly 1`] = `
<span
  className=""
>
  Test
</span>
`;
